scripts/Test-Signatures.ps1

```diff
@@ -0,0 +1,209 @@
+[CmdletBinding()]
+param (
+    [Parameter()]
+    [String]
+    $Path = "$PSScriptRoot\packages",
+
+    [Parameter()]
+    [Switch]
+    $CIBuild
+)
+
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+
+# Allowed certificate subjects
+$global:ValidSubjects = @( `
+    "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", `
+    "CN=Microsoft Corporation, OU=MOPR, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" `
+);
+
+# Allowed certificate thumbprints
+$global:ValidThumbprints = @( `
+    "B9EAA034C821C159B05D3521BCF7FEB796EBD6FF", `
+    "98ED99A67886D020C564923B7DF25E9AC019DF26", `
+    "5EAD300DC7E4D637948ECB0ED829A072BD152E17", `
+    "67B1757863E3EFF760EA9EBB02849AF07D3A8080", `
+    "9DC17888B5CFAD98B3CB35C1994E96227F061675", `
+    "62009AAABDAE749FD47D19150958329BF6FF4B34", `
+    "3BDA323E552DB1FDE5F4FBEE75D6D5B2B187EEDC" `
+)
+
+# Allow-list for unsigned binaries
+$global:UnsignedAllowList = @{
+    # "PackageId" = relative paths
+    "Microsoft.TestPlatform" = @( `
+        ".\tools\net451\Common7\IDE\Extensions\TestPlatform\Newtonsoft.Json.dll" `
+    )
+
+    "Microsoft.TestPlatform.CLI" = @( `
+        ".\contentFiles\any\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\contentFiles\any\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+    
+    "Microsoft.TestPlatform.Portable" = @( `
+        ".\tools\net451\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\TestHost\Newtonsoft.Json.dll", `
+        ".\tools\netcoreapp2.1\Newtonsoft.Json.dll"
+    )
+}
+
+
+$global:Depth = 0;
+
+function Get-RelativePath($path, $relativeTo) {
+    Push-Location

```

Comment:
Just `Join-Path $relativeTo $path` and then resolve it? No need to jump around directories.

Reply:
This function actually returns the relative path. This was the safest way of doing that I could think of.

Example:
```
      Path: c:\git\package\tools\netcoreapp2.1\Newtonsoft.Json.dll
RelativeTo: c:\git\package
    Output: .\tools\netcoreapp2.1\Newtonsoft.Json.dll
```

Reply:
Ah okay, Use try finally for the pushing and popping then?